recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -0,0 +1,136 @@
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def _detect_compilers(self):
+        cmake_cache = tools.load(os.path.join(self.build_folder, "CMakeCache.txt"))
+        cc = re.search("CMAKE_C_COMPILER:FILEPATH=(.*)", cmake_cache)[1]
+        cxx = re.search("CMAKE_CXX_COMPILER:FILEPATH=(.*)", cmake_cache)[1]
+        return cc, cxx
+
+    def build(self):        
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        # build aws-cpp-sdk-cdi
+        cmake = self._configure_cmake()
+        cmake.build()
+
+        autotools = self._configure_autotools()
+        with tools.chdir(self._source_subfolder):
+            # configure autotools to find aws-cpp-sdk-cdi
+            autotools.include_paths.append(os.path.join(self.build_folder, self._source_subfolder, "aws-cpp-sdk-cdi", "include"))
+            autotools.library_paths.append(os.path.join(self.build_folder, "lib"))
+            autotools.libs.append("aws-cpp-sdk-cdi")
+
+            vars = autotools.vars
+            cc, cxx = self._detect_compilers()
+            vars["CC"] = cc
+            vars["CXX"] = cxx
+            if self.settings.build_type == "Debug":
+                vars["DEBUG"] = "y"
+
+            args = ["require_aws_sdk=no"]
+
+            autotools.make(target="libsdk", vars=vars, args=args)
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        config = "debug" if self.settings.build_type == "Debug" else "release"
+        self.copy(pattern="*", dst="lib", src=os.path.join(self._source_subfolder, "build", config, "lib"))
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        cppSdk = self.cpp_info.components["aws-cpp-sdk-cdi"]
+        cppSdk.libs = ["aws-cpp-sdk-cdi"]
+
+        cppSdk.requires = ["aws-sdk-cpp::monitoring", "aws-libfabric::aws-libfabric"]
+        
+        cppSdk.set_property("cmake_file_name", "aws-cpp-sdk-cdi")
+        cppSdk.set_property("cmake_target_name", "aws-cpp-sdk-cdi")
+        cppSdk.set_property("pkg_config_name", "aws-cpp-sdk-cdi")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        # TODO: Remove the namespace on CMake targets
+        cppSdk.names["cmake_find_package"] = "aws-cpp-sdk-cdi"
+        cppSdk.names["cmake_find_package_multi"] = "aws-cpp-sdk-cdi"
+        cppSdk.names["pkg_config"] = "aws-cpp-sdk-cdi"
```

Comment:
it's not consistent.

Reply:
I've set both to be `AWS::aws-cpp-sdk-cdi` similar to the aws-sdk-cpp package. 
I couldn't anything existing practice upstream.

Reply:
If you don't know how to deduce imported targets from upstream CMakeLists (not always obvious), you inspect packaged files locally:

- comment out the deletion of cmake files during packaging
- inspect `lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets.cmake` in package folder and look for `add_library()`

From what I can see in build log, `cmake_file_name` is `aws-cpp-sdk-cdi`:

```
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-targets-release.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-config.cmake
-- Installing: /home/conan/w/BuildSingleReference/.conan/data/aws-cdi-sdk/2.2.0/_/_/package/7e1489d0a9b824ecd6e0f2c6473d3bdf325c0375/lib/cmake/aws-cpp-sdk-cdi/aws-cpp-sdk-cdi-config-version.cmake
```

Reply:
I see. Very confusing build system: why do they mix CMakeLists to build aws-cpp-sdk-cdi  & raw Makefile to build aws-cdi-sdk?

Reply:
I guess different people worked on each component